conda_smithy/feedstock_io.py

```diff
@@ -82,11 +82,13 @@ def remove_file(filename):
     if repo:
         repo.index.remove([filename])
 
-    os.remove(filename)
+    Path(filename).unlink()
 
-    dirname = os.path.dirname(filename)
-    if dirname and not os.listdir(dirname):
-        os.removedirs(dirname)
+    dirname = Path(filename).parent
+    if dirname.exists() and not any(dirname.iterdir()):
+      while dirname.exists() and not any(dirname.iterdir()):
```

Comment:
The `if` statement here looks redundant as it has the same condition as the `while` loop inside it. Also Python standard indentation is 4 spaces.